### DISCUSSION TITLE
Shared state tutorial

### DISCUSSION THREAD

**Discussion by Lun4m** [tokio-rs/tokio#6518] [Q&A] [answered]
I'm a bit confused by the second snippet for the sharded DB. Where does the `hash` function come from? Or is it supposed to be pseudocode?
```rust
let shard = db[hash(key) % db.len()].lock().unwrap();
shard.insert(key, value);
```

**Comment by Darksonn** [comment, marked as answer]
You'll have to choose a hash function that is appropriate for whatever the type of `key` is. There are many different hash functions out there, and the example will work for whichever choice you make.

**Reply by Lun4m** [reply]
Thanks for the response! I see, yeah, I guess I was confusing that function with the default HashMap hash function :sweat_smile:
